docs/source/getting_started/benchmarks/imagenette_benchmark.py

```diff
@@ -465,11 +475,63 @@ def configure_optimizers(self):
         scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, max_epochs)
         return [optim], [scheduler]
 
-model_names = ['MoCo_256', 'SimCLR_256', 'SimSiam_256', 'BarlowTwins_256', 
-               'BYOL_256', 'NNCLR_256', 'NNSimSiam_256', 'NNBYOL_256']
 
-models = [MocoModel, SimCLRModel, SimSiamModel, BarlowTwinsModel, 
-          BYOLModel, NNCLRModel, NNSimSiamModel, NNBYOLModel]
+class SwaVModel(BenchmarkModule):
+    def __init__(self, dataloader_kNN, num_classes):
+        super().__init__(dataloader_kNN, num_classes)
+        # create a ResNet backbone and remove the classification head
+        resnet = torchvision.models.resnet18()
+        last_conv_channels = list(resnet.children())[-1].in_features
+        self.backbone = nn.Sequential(
+            *list(resnet.children())[:-1],
+            nn.Conv2d(last_conv_channels, num_ftrs, 1),
+        )
+
+        self.projection_head = SwaVProjectionHead(512, 512, 128)
+        self.prototypes = SwaVPrototypes(128, 512) # use 200 prototypes
+
+        self.criterion = lightly.loss.SwaV()
+
+    def forward(self, x):
+        x = self.backbone(x).flatten(start_dim=1)
+        x = self.projection_head(x)
+        x = nn.functional.normalize(x, dim=1, p=2)
+        return self.prototypes(x)
+
+    def training_step(self, batch, batch_idx):
+
+        with torch.no_grad():
+            self.prototypes.normalize_weights()
+
+        multi_crops, _, _ = batch
```

Comment:
I'm not a big fan of how we handle the whole data input. It's now a hack in a hack. One big list with images of different sizes...

If you want to keep this in here I would add a few comments on what exactly `multi_crops` is (e.g. shape). Also that the first 2 elements are the high-res ones and the other 4? are the low-res ones.

Reply:
After our discussion I will simply add comments here to explain everything. This will likely change again with the refactoring of the collate function.